modules/store.js

```diff
@@ -0,0 +1,466 @@
+                    return rp({
+                        method: 'POST',
+                        url: this.storeLocator,
+                        body: this.generateStoreRequestBody(googleBody.results[0].geometry, count),
+                        json: true
+                    }).then(wizardsBody => this.generateStoreEmbed(wizardsBody.d.Results, googleBody.results[0]),
+                        err => this.generateErrorEmbed(err, "Couldn't retrieve stores from Wizards."));
+                } else {
+                    // use cached stores to calculate closest stores
+                    const location = {
+                        lat: googleBody.results[0].geometry.location.lat,
+                        lng: googleBody.results[0].geometry.location.lng
+                    }
+                    // create a copy of stores and sort it by distance to our location
+                    let results = this.stores.slice();
+                    results.sort((a, b) =>
+                        this.getDistance(a.Address.Latitude, a.Address.Longitude, location.lat, location.lng) -
+                        this.getDistance(b.Address.Latitude, b.Address.Longitude, location.lat, location.lng));
+                    return this.generateStoreEmbed(results.slice(0, count), googleBody.results[0]);
+                }
+            } else {
+                return this.generateErrorEmbed(null, `Location \`${location}\` not found.`);
+            }
+        }, err => this.generateErrorEmbed(err, "Couldn't retrieve location from Google."));
+    }
+
+    generateStoreEmbed(stores, googleResult) {
+        const fields = [];
+        const googleStaticMap = [this.googleStaticMap];
+        stores.forEach(({Address, Organization, IsStore}) => {
+            // only take the first 6 stores
+            if (!IsStore || fields.length > 5) return;
+            // calculate distance in KM
+            const distance = this.getDistance(
+                Address.Latitude, Address.Longitude,
+                googleResult.geometry.location.lat, googleResult.geometry.location.lng);
+
+            fields.push({
+                name: `${fields.length + 1}) ${Address.Name} (${Math.round(distance)}km)`,
+                value: this.generateStoreDescription(Address, Organization),
+                inline: true
+            });
+
+            googleStaticMap.push(`${this.googleStaticMapStandardMarkup}label:${fields.length}|${Address.Line1} `+
+                `${Address.PostalCode} ${Address.City} ${Address.CountryName}`);
+        });
+        // fetch map from google
+        return rp({url: encodeURI(googleStaticMap.join("")), encoding: null}).then(body =>
+            new Discord.RichEmbed({
+                title: `Stores closest to ${googleResult.formatted_address}`,
+                description: `:link: [Wizards Store Locator results](http://locator.wizards.com/#brand=magic&a=search&p=${encodeURIComponent(googleResult.formatted_address)}&massmarket=no)`,
```

Comment:
Maybe store the URL in a variable like the others?

Reply:
Good point, fixed. :D